File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/DevolucionProveedorResponse.java
package com.farmacia.sanidadbackend.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class DevolucionProveedorResponse {

    private Long id;
    private String numeroDevolucion;
    private Long loteId;
    private String numeroFacturaLote; // lote.factura
    private Long proveedorId;
    private String proveedorNombre;
    private String proveedorTelefono;
    private String proveedorEmail;

    private Long solicitadoPorId;
    private String solicitadoPorNombre;

    private Long aprobadoPorId;
    private String aprobadoPorNombre;

    private String estado;
    private String motivo;
    private LocalDateTime fechaSolicitud;
    private LocalDateTime fechaAprobacion;

    private List<DevolucionProveedorDetalleResponse> detalles;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/DevolucionProveedorResponse.kt
package com.sanidad.movil.data.remote.dto

data class DevolucionProveedorResponse(
    val id: Long,
    val numeroDevolucion: String,
    val loteId: Long,
    val numeroFacturaLote: String?,
    val proveedorId: Long,
    val proveedorNombre: String,
    val proveedorTelefono: String?,
    val proveedorEmail: String?,
    val solicitadoPorId: Long,
    val solicitadoPorNombre: String,
    val aprobadoPorId: Long?,
    val aprobadoPorNombre: String?,
    val estado: String,
    val motivo: String?,
    val fechaSolicitud: String,
    val fechaAprobacion: String?,
    val detalles: List<DevolucionProveedorDetalleResponse>
)